src/App.js

```diff
@@ -3,11 +3,57 @@ import './App.css'
 
 
 class App extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      color: '#000000',
+      usedColors: []
+    }
+
+    this.changeColor = this.changeColor.bind(this)
+  }
+
+  getNewColor() {
+    return '#' + Math.floor(Math.random()*16777215).toString(16)
+  }
+
+  changeColor() {
+    const { usedColors } = this.state
+    const newColor = this.getNewColor()
+
+    if (usedColors.includes(newColor)) {
+      // Beware stack overflows here...
+      this.changeColor()
```

Comment:
Is there a way to do this so that we wouldn't have a stack overflow? If so, please update